server/fleet/datastore.go

```diff
@@ -1253,7 +1253,11 @@ type Datastore interface {
 	InsertMDMConfigAssets(ctx context.Context, assets []MDMConfigAsset) error
 
 	// GetMDMConfigAssetsByName returns the requested config assets.
-	GetMDMConfigAssetsByName(ctx context.Context, assetNames []MDMAssetName) ([]MDMConfigAsset, error)
+	//
+	// - If it doesn't find any assests, it returns a nil map
+	// - If it only finds some of the assets, it returns the assets found
+	//   and ErrPartialResult as the error
+	GetMDMConfigAssetsByName(ctx context.Context, assetNames []MDMAssetName) (map[MDMAssetName]MDMConfigAsset, error)
```

Comment:
does this make sense? or should we always return the error? rationale: in some cases the caller only wants to know if there records or not.

for those cases, if we always return the error, the caller needs to guard against it and do a `len` check as well.

the problem with this is that it might open unexpected edge cases, for example: if you only pass one asset, and we couldn't find it then it's a Schrödinger's asset 😓

Reply:
I feel like it should always return the error? `nil, ErrPartialResult` in that case, and the caller should check with `errors.Is` (a bit like the sql.ErrNoRows thing)? One thought I had this morning, since this is different from the other "get stuff from list of names or ids" that don't return an error if it doesn't find some, how would you feel about renaming the method to `MustGetMDMConfigAssetsByName` (kind of following Go's approach of `Must` prefix, although we don't panic but we return an error)? Just trying to see if we can make that different behaviour jump out a bit more, instead of just relying on docs.

Reply:
Or perhaps `GetAllMDMConfigAssetsByName`?

Reply:
that makes a lot of sense, thanks! I went in and did both (the rename) and the change of behavior.

I went with `GetAllMDMConfigAssetsByName` but happy to change it (it's just a call to the LSP to rename, no much work)